images/docker-build.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+CYAN='\033[0;36m'
+NC='\033[0m'
+
+repository="quay.io/pusher"
+
+for arg in "$@"; do
+  case ${arg%%=*} in
+    "--version")
+      version="${arg##*=}"
+      ;;
+   "--build-root")
+      build_root="${arg##*=}"
+      ;;
+    "--help")
+      printf "${GREEN}$0${NC}\n"
+      printf "  available options:\n"
+      printf "  --version=${BLUE}<image version tag>${NC}\n"
+      printf "  --build-root=${BLUE}<docker build root>${NC}\n"
+      exit 0
+      ;;
+    *)
+      echo "Unknown option: $arg"
+      exit 2
+      ;;
+    esac
+done
+
+##############################################################################
+#
+# Use build cache from the given version if it exists, else use the latest tag
+#
+##############################################################################
+curl --location --fail --silent --header "Accept: application/vnd.docker.distribution.manifest.v2+json" \
+"http://quay.io/v2/pusher/$build_root/manifests/$version" &> /dev/null
+if [ $? != 0 ]; then
+  # Tag does not exist, cache from latest
+  cache_tag="latest"
+else
+  # Tag exists, cahce from this version
+  cache_tag=$version
+fi
+
+###########################################
+#
+# Check if the image is a multi stage build
+#
+###########################################
+lines=$(cat ${build_root}/Dockerfile | grep 'FROM')
```

Comment:
```suggestion
lines=$(grep 'FROM .* AS' ${build_root}/Dockerfile)
```

And then you won't need the if statement in the while loop :)